base/c.jl

```diff
@@ -75,11 +75,7 @@ cconvert(::Type{Cstring}, s::String) =
 cconvert(::Type{Cstring}, s::AbstractString) =
     cconvert(Cstring, String(s)::String)
 
-function cconvert(::Type{Cwstring}, s::AbstractString)
-    v = transcode(Cwchar_t, String(s).data)
-    !isempty(v) && v[end] == 0 || push!(v, 0)
-    return v
-end
+cconvert(::Type{Cwstring}, s::AbstractString) = cwstring(s)
```

Comment:
I reverted this bit. Basically we have the same logic defined twice (once for the implicit `cconvert`, `unsafe_convert` via `Cwstring`, and once for the explicit `cwstring` function. But their precise order (i.e. when they do the checks) differs, so I decided to leave it.